### ISSUE TITLE
ClientNackFilter leads to "Too many headers" error

### ISSUE THREAD

**Issue by red1ynx** [twitter/finagle#721]
`ClientNackFilter` appends "finagle-http-retryable-request:" header every time for failed requests which leads to "Too many headers" error on server-side.

### Expected behavior

The server below should accept all incoming requests. "finagle-http-retryable-request:" should appear in header only once.

### Actual behavior
The number of "finagle-http-retryable-request:" headers increases over time.
The first request:
```
X-B3-SpanId: f55e78e28ee58b6e
Finagle-Ctx-com.twitter.finagle.Retries: 0
Accept: application/json
finagle-http-retryable-request:
Host: localhost:8080
Content-Length: 14
Content-Type: application/json
X-B3-TraceId: f55e78e28ee58b6e


{"foo": "bar"}
```

10th request:
```
X-B3-SpanId: 62609d73c8a0283c
Finagle-Ctx-com.twitter.finagle.Retries: 0
Accept: application/json
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
finagle-http-retryable-request:
Host: localhost:8080
Content-Length: 14
Content-Type: application/json
X-B3-TraceId: 62609d73c8a0283c


{"foo": "bar"}
```

And the server generates "code 431, message Too many headers" for the last several requests.
```
127.0.0.1 - - [29/Aug/2018 23:56:06] code 431, message Too many headers
127.0.0.1 - - [29/Aug/2018 23:56:06] "POST / HTTP/1.1" 431 -
127.0.0.1 - - [29/Aug/2018 23:56:06] code 431, message Too many headers
127.0.0.1 - - [29/Aug/2018 23:56:06] "POST / HTTP/1.1" 431 -
127.0.0.1 - - [29/Aug/2018 23:56:06] code 431, message Too many headers
127.0.0.1 - - [29/Aug/2018 23:56:06] "POST / HTTP/1.1" 431 -
127.0.0.1 - - [29/Aug/2018 23:56:06] code 431, message Too many headers
```

### Steps to reproduce the behavior

Finagle version: `18.7.0`
Ammonite script of client:
```
import $ivy.`com.twitter::finagle-http:18.7.0`
import com.twitter.finagle._, com.twitter.util._
import com.twitter.finagle.http._
import com.twitter.io.Buf
import java.net.URL

val client = Http.client.newService("localhost:8080")

val req = RequestBuilder()
  .url(new URL("http://localhost:8080"))
  .setHeader("Content-Type", "application/json")
  .setHeader("Accept", "application/json")
  .buildPost(Buf.Utf8("""{"foo": "bar"}"""))

(1 to 100).foreach(_ => {
  client(req)
  Thread.sleep(100)
})
```

Server: https://gist.github.com/bjorm/7248293 , port 8080.

**Comment by mosesn** [comment]
eek, good catch.  I think the bug is that https://github.com/twitter/finagle/blob/develop/finagle-http/src/main/scala/com/twitter/finagle/http/filter/ClientNackFilter.scala#L34 should set `set` instead of `add`.  Do you want to tackle it?  If you're busy, we can grab it.

**Comment by bryce-anderson** [comment]
I wrote this bug so I can put up a patch. Thanks for reporting it @red1ynx.

**Comment by bryce-anderson** [comment]
This should be fixed by commit b49a29790293bb157e943a3c783c7d34062bfabe. Thanks for the bug report @red1ynx!
